bot/plugins/events.py

```diff
@@ -0,0 +1,14 @@
+from __future__ import annotations
+
+import crescent
+import hikari
+
+from bot.core.event import handle_event
+
+plugin = crescent.Plugin("custom-events")
+
+
+@plugin.include
+@crescent.event
+async def on_message(event: hikari.GuildMessageCreateEvent) -> None:
+    await handle_event(event)
```

Comment:
I think events should be passed through their own serialization function before we start handling them

Reply:
That could become very hard. Models have a lot of data. You're welcome to try though